src/Vault.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: BSD Clause-3
+pragma solidity ^0.8.24;
```

Comment:
overall approach  looks good 
 
 1.starting off with this mixed ERC4626 and multi-asset vault interface
 2. build the rest of the logic
 
 figure out later what separate modules we need to add based on what is common and what's different or looks like it needs to be upgraded